src/dataEntities/appointment.ts

```diff
@@ -231,31 +232,32 @@ export class Appointment {
     /**
      * Parse an appointment and check property types.
      * @param obj
+     * @param log Logger to be used in case of failures
      */
-    public static parse(obj: any) {
+    public static parse(obj: any, log?: Logger) {
```

Comment:
perhaps a default parameter `log?: Logger = logger` would be simpler, instead of using `(log || logger)` later?

Reply:
Yeah, updated.

Reply:
The `parse` function still doesn't have the default parameter. Moreover, once there is the default parameter, you can replace `(log || logger)` with just `log`.